src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -3985,7 +3985,14 @@ module EstablishTypeDefinitionCores =
                 match fields' with 
                 | rf :: _ -> errorR (Error(FSComp.SR.tcInterfaceTypesAndDelegatesCannotContainFields(), rf.Range))
                 | _ -> ()
-
+            
+            let primaryConstructorInDelegateCheck(tyCon: Tycon, implicitCtorSynPats : SynSimplePats option) =
+                match implicitCtorSynPats with 
+                | None -> ()
+                | Some spats ->
+                    if tyCon.IsFSharpDelegateTycon then
```

Comment:
Is this function going to be called on non-delegate type definitions? If not, then this check is always true and is not needed.

Reply:
Yeah that check will always be true . I have removed the  check